pkg/loadbalancer/experimental/lbmaps.go

```diff
@@ -242,6 +392,21 @@ func (r *BPFLBMaps) Stop(cell.HookContext) error {
 	return errors.Join(errs...)
 }
 
+// iterateMap iterates over a BPF map, yielding new keys and values. Similar to
+// ebpf.IterateWithCallback but allocates new key & value instead of reusing.
+func iterateMap(m *ebpf.Map, newPair func() (any, any), cb func(any, any)) error {
+	entries := m.Iterate()
```

Comment:
The original `ebpf.IterateWithCallback` calls `m.lock.RLock()`. Why is it not needed here?

Reply:
No need, it's safe to call `Iterate` concurrently.

Reply:
> I don't know why ebpf.Map has a lock here.

Perhaps the reason is to obtain a consistent state in case write happens concurrently?

Reply:
I see, thanks!